### DISCUSSION TITLE
Is there support for nested object properties with x-model?

### DISCUSSION THREAD

**Discussion by ProNotion** [alpinejs/alpine#3428] [1. Help] [answered]
I'm fairly new to AlpineJs and all of the docs I've read give examples of data models using properties at the top level. 

However, as an example I have the following x-data object:

```
{ 
  studentMember: false,
  application: { 
    studentMember: false
  }
}
```

My checkbox:
```
<input type="checkbox" value="true" x-model="application.studentMember"/>
```

I have a checkbox on the page as shown above and if it is bound to `application.studentMember` it does not update yet if it is bound to `studentMember` it works as expected. Is this expected? It seems wasteful and unnecessary to have to duplicate properties at the top level and watch them for changes in order to update the nested object properties.

**Comment by ekwoka** [comment]
Yes. it is supported. You can even model into an array by index.

Internally, alpine essentially makes two functions

one similar to 

```js
(val) => data.application.studentMember = val
```

and 

```js
() => data.application.studentMember
```

And uses those to get and set the data.

**Reply by ProNotion** [reply]
In that case, why in my example might it not be reacting to the change in the nested property but does with the one that is not nested?

**Reply by SimoTod** [reply]
You probably have an error somewhere else, it seems to work for me: https://codepen.io/SimoTod/pen/abadZOR

**Reply by ekwoka** [reply]
I have no idea. I'd expect something to do with the code not shown.

**Comment by ProNotion** [comment, marked as answer]
Thanks for your patience. Your example works and when I tried to replicate a simple example based on my implementation that also worked so I went back to my code and had it render out the value and sure enough it was already working! Turns out the Alpine.js devtools Chrome extension I am using and where I was watching the current value does not update when I toggle the checkbox on/off.

**Reply by MathiasSven** [reply]
Same thing for me, I am just learning Alpine and thought I was getting it wrong, but it is the extension that had a bug! It is really early days, but I wonder if the AlpineJs team has some form of devtool like that in mind for the future.

**Reply by ekwoka** [reply]
Theres only one person on the "team", and then some others that help out.

It's on my personal roadmap, but there is quite a few other things on my list first (I try to focus on the things as tools that are most useful to me personally and the current extension is often good enough for the places I might need it) like component testing, and smarter component design patterns.
